tasks/cluster.yml

```
@@ -0,0 +1,47 @@
+---
+- name: Set redis cluster command
+  set_fact:
+    redis_cluster_command: >-
+      {%- set redis_instances = [] -%}
+      {%- for host in ansible_play_hosts_all -%}
+        {%- for i in redis_configurations -%}
+          {%- set host_ip = hostvars[host]['ansible_default_ipv4']['address'] | default(ansible_all_ipv4_addresses[0]) -%}
+          {%- if i['redis_tls_port'] is defined -%}
+            {%- set host_port = i['redis_tls_port'] -%}
+          {%- else -%}
+            {%- set host_port = i['redis_port'] -%}
+          {%- endif -%}
+          {{ redis_instances.append( host_ip+ ':' +host_port ) }}
+        {%- endfor -%}
+      {%- endfor -%}
+      redis-cli --cluster create {{ redis_instances | join(' ') }} --cluster-replicas {{ redis_masters_replicas }} -a {{ redis_requirepass }} --cluster-yes {% if redis_configurations[0]['redis_tls_cluster'] is defined and redis_configurations[0]['redis_tls_cluster'] == 'yes' %} --tls --cert {{ redis_configurations [0]['redis_tls_cert_file'] }} --key {{ redis_configurations [0]['redis_tls_key_file'] }} --cacert {{ redis_configurations [0]['redis_tls_ca_cert_file'] }} {% endif %}
```

Since you're already using YAML multiline, you could put every option in a separate line. This would help with changing the command later if there are any new changes to be done.
```suggestion
      redis-cli
      --cluster
      create {{ redis_instances | join(' ') }}
      --cluster-replicas {{ redis_masters_replicas }}
      ...
```
(I'm not writing the whole command, because 1. I'm lazy and 2. I don't want to hijack your line and mark it as my own. :-) )